custom-instrumentation/instrument-datastore/instrumentation.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+const datastorePath = require.resolve('./simple-datastore')
+
+newrelic.instrumentDatastore({
+  // The path to our datastore module
+  absolutePath: datastorePath,
+  // The name of the datastore module
+  moduleName: 'simple-datastore',
+  // The function that will instrument the datastore
+  // if it is successfully required.
+  onRequire: instrumentSimpleDatastore,
+  // The function that will be called if there is
+  // an error with the instrumentation.
+  onError: function myErrorHandler (err) {
+    console.error(err.message, err.stack)
+    process.exit(1)
+  },
+})
+
+function instrumentSimpleDatastore (shim, SimpleDatastore, moduleName) {
+  console.log(`[NEWRELIC] instrumenting ${moduleName}`)
+  const proto = SimpleDatastore.prototype
+
+  // This is required to set the datastore name in the transaction trace
+  shim.setDatastore(moduleName)
+
+  // This function instruments our datastore operations.
+  // You may provide a OperationSpec or SegmentFunction for the opSpec.
+  // We chose SegmentFunction here because we want to know the name of the operation.
+  shim.recordOperation(proto, ['connect', 'close'], function (shim, fn, name, args) {
```

Comment:
you're running connect in the example outside a transaction so this isn't doing anything